test/e2e/httpproxy/cookie_rewrite_test.go

```diff
@@ -0,0 +1,827 @@
+				attributePairs := strings.Split(v, ";")
+				for _, p := range attributePairs {
+					split := strings.Split(strings.TrimSpace(p), "=")
+					if len(split) == 0 {
+						continue
+					}
+					attributeName := split[0]
+					attributeValue := ""
+					if len(split) > 1 {
+						attributeValue = split[1]
+					}
+					attributes[attributeName] = attributeValue
+				}
+			}
+		}
+	}
+	return attributes
+}
+
+func checkReturnedSetCookieHeader(headers http.Header, cookieName, cookieValue, path, domain, sameSite string, secure bool, additionalAttrs map[string]string) {
+	f.T().Helper()
+
+	attributes := parseCookieAttributes(headers, cookieName)
+
+	// If cookie value is empty we ignore it (e.g. a generated cookie we don't know the value of)
+	if len(cookieValue) > 0 {
+		assert.Equal(f.T(), cookieValue, attributes[cookieName])
+	}
+	if len(path) > 0 {
+		assert.Equal(f.T(), path, attributes["Path"])
+	} else {
+		assert.NotContains(f.T(), attributes, "Path")
+	}
+	if len(domain) > 0 {
+		assert.Equal(f.T(), domain, attributes["Domain"])
+	} else {
+		assert.NotContains(f.T(), attributes, "Domain")
+	}
+	if len(sameSite) > 0 {
+		assert.Equal(f.T(), sameSite, attributes["SameSite"])
+	} else {
+		assert.NotContains(f.T(), attributes, "SameSite")
+	}
+	if secure {
+		assert.Contains(f.T(), attributes, "Secure")
+	} else {
+		assert.NotContains(f.T(), attributes, "Secure")
+	}
+
+	for a, v := range additionalAttrs {
+		assert.Equal(f.T(), v, attributes[a])
+	}
+}
+
+func requestSetCookieHeader(https bool, host, route string, setCookieValues ...string) http.Header {
+	f.T().Helper()
+	opts := []func(*http.Request){
+		e2e.OptSetHeaders(map[string]string{
+			"X-ECHO-HEADER": "Set-Cookie:" + strings.Join(setCookieValues, ", Set-Cookie:"),
+		}),
+	}
+
+	var res *e2e.HTTPResponse
+	var ok bool
+	if https {
+		res, ok = f.HTTP.SecureRequestUntil(&e2e.HTTPSRequestOpts{
+			Path:        route,
+			Host:        host,
+			RequestOpts: opts,
+			Condition:   e2e.HasStatusCode(200),
+		})
+	} else {
+		res, ok = f.HTTP.RequestUntil(&e2e.HTTPRequestOpts{
+			Path:        route,
+			Host:        host,
+			RequestOpts: opts,
+			Condition:   e2e.HasStatusCode(200),
+		})
+	}
+
+	require.NotNil(f.T(), res)
+	require.Truef(f.T(), ok, "expected 200 response code, got %d", res.StatusCode)
+
+	return res.Headers
+}
+
+func deployEchoServer(t require.TestingT, c client.Client, ns, name string) {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app.kubernetes.io/name": name},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app.kubernetes.io/name": name},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "echo",
+							Image: "docker.io/ealen/echo-server:0.5.1",
```

Comment:
will need to use this alternative echo server, or get custom features into the ingress-conformance one

might want to push this into `fixtures.go` so it gets loaded into the test cluster before tests run